renderer/src/shared/Hint/positioning.ts

```diff
@@ -0,0 +1,278 @@
+              );
+
+              Object.assign(floatingRef.current.style, {
+                left: `${x + shiftX}px`,
+              });
+              Object.assign(SVGRef.current.style, {
+                left: `${x + hintWidth / 2}px`,
+              });
+            }
+            if (position.diagonalPosition === "left") {
+              HorizontalLine.setAttribute("x1", `${shiftX + hintWidth / 2}`);
+              HorizontalLine.setAttribute("x2", `${hintWidth}`);
+
+              VerticalLine.setAttribute("x1", `${shiftX + hintWidth / 2}`);
+              VerticalLine.setAttribute("x2", `${shiftX + hintWidth / 2}`);
+
+              TriangleRef.current.setAttribute(
+                "points",
+                `${shiftX + hintWidth / 2 - 5}, ${hintHeight - 10} 
+                ${shiftX + hintWidth / 2}, ${hintHeight} 
+                ${shiftX + hintWidth / 2 + 5},  ${hintHeight - 10} `
+              );
+
+              Object.assign(floatingRef.current.style, {
+                left: `${x - shiftX}px`,
+              });
+              Object.assign(SVGRef.current.style, {
+                left: `${x - shiftX}px`,
+              });
+            }
+            break;
+
+          case "right":
+            Object.assign(floatingRef.current.style, {
+              left: `${x + shiftX}px`,
+            });
+            Object.assign(SVGRef.current.style, {
+              top: `${y - shiftY}px`,
+              left: `${x}px`,
+              width: `${hintWidth / 2 + shiftX + 1}`,
+            });
+
+            HorizontalLine.setAttribute("x1", `${0}`);
+            HorizontalLine.setAttribute("x2", `${hintWidth / 2 + shiftX}`);
+            HorizontalLine.setAttribute("y1", `${hintHeight / 2 + shiftY}`);
+            HorizontalLine.setAttribute("y2", `${hintHeight / 2 + shiftY}`);
+
+            VerticalLine.setAttribute("x1", `${hintWidth / 2 + shiftX}`);
+            VerticalLine.setAttribute("x2", `${hintWidth / 2 + shiftX}`);
+            VerticalLine.setAttribute("y1", `${hintHeight / 2 + shiftY}`);
+
+            TriangleRef.current.setAttribute(
+              "points",
+              `${10}, ${hintHeight / 2 + shiftY - 5} 
+              ${0}, ${hintHeight / 2 + shiftY} 
+              ${10},  ${hintHeight / 2 + shiftY + 5} `
+            );
+
+            if (position.diagonalPosition === "top") {
+              VerticalLine.setAttribute("y2", `${hintHeight}`);
+
+              Object.assign(floatingRef.current.style, {
+                top: `${y - shiftY}px`,
+              });
+            } else if (position.diagonalPosition === "bottom") {
+              VerticalLine.setAttribute("y2", `${hintHeight / 2 + shiftY * 2}`);
+
+              Object.assign(SVGRef.current.style, {
+                height: `${hintHeight + shiftY * 2}`,
+              });
+              Object.assign(floatingRef.current.style, {
+                top: `${y + shiftY}px`,
+              });
+            }
+            break;
+
+          case "bottom":
```

Comment:
Could you please create a separate constant for the bottom, top, right, left?
Example `export const PLACEMENTS = { TOP: "top" }`